engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/event/IntermediateThrowEventProcessor.java

```diff
@@ -186,4 +197,48 @@ public void onComplete(
               failure -> incidentBehavior.createIncident(failure, completing));
     }
   }
+
+  private class EscalationIntermediateThrowEventBehavior implements IntermediateThrowEventBehavior {
+    @Override
+    public boolean isSuitableForEvent(final ExecutableIntermediateThrowEvent element) {
+      return element.isEscalationThrowEvent();
+    }
+
+    @Override
+    public void onActivate(
+        final ExecutableIntermediateThrowEvent element, final BpmnElementContext activating) {
+      final var escalation = element.getEscalation();
+      ensureNotNull("escalation", escalation);
+
+      final var escalationCode = escalation.getEscalationCode();
+      ensureNotNullOrEmpty("escalationCode", escalationCode);
+
+      final var activated = stateTransitionBehavior.transitionToActivated(activating);
+      final Optional<CatchEventTuple> escalationCatchEvent =
+          eventPublicationBehavior.findEscalationCatchEvent(escalationCode, activating);
+
+      final EscalationRecord record = new EscalationRecord();
+      record.setCatchElementId(BufferUtil.wrapString(""));
+      record.setThrowElementId(element.getId());
+      record.setEscalationCode(BufferUtil.bufferAsString(escalationCode));
+      record.setProcessInstanceKey(activated.getProcessInstanceKey());
+      final boolean canBeCompleted =
+          eventPublicationBehavior.throwEscalationEvent(record, escalationCatchEvent);
+
+      if (canBeCompleted) {
+        stateTransitionBehavior.completeElement(activated);
+      }
+    }
```

Comment:
❌ I think we put too much responsibility in this method. A bunch of this logic will be duplicated when implementing escalation end event. For example this will also have to find the catch event and create the `EscalationRecord`.

I'd like to propose a slightly different propose in which we delegate this functionality to the `BpmnEventPublicationBehavior`.

We can use a fire-and-forget method here. At this point we don't care if there is a catch event or not. The beauty of escalations is that if it doesn't get caught we just continues processing. This means we can greatly simplify this method. This method would then look something like this:
```java
final var activated = stateTransitionBehavior.transitionToActivated(activating);
eventPublicationBehavior.throwEscalationEvent(element, activated);
stateTransitionBehavior.completeElement(activated);
```

The `throwEscalationEvent` will then create the record. It will also try and find a catch event. If it can find one it will activate it and write the `ESCALATED` event. If it can't find any it will write the `NOT_ESCALATED` event.

Reply:
💭 I see that you do need to check if we can complete the element depending on whether or not the event is interrupting, so the `throwEscalationEvent` method would still return a boolean which we can check in the processor.